csrc/fusion_segmenter.cpp

```diff
@@ -1969,7 +1970,23 @@ std::pair<IrCloner, std::unique_ptr<Fusion>> SegmentedFusion::makeFusion(
   return std::make_pair(complete_to_segment_map, std::move(fusion_segment));
 }
 
-std::unique_ptr<SegmentedFusion> SegmentCandidateFinder::segment(
+/*static*/ std::unique_ptr<SegmentedFusion> SegmentCandidateFinder::segment(
+    std::unique_ptr<Fusion> fusion,
+    const KernelArgumentHolder* inputs,
+    const SegmentCandidateFinderOptions& options) {
+  if (isDebugDumpEnabled(DebugDumpOption::FusionSegments)) {
+    const auto& communicator = Communicator::getInstance();
+    if (!communicator.is_available() || communicator.local_rank() == 0) {
```

Comment:
Why using `local_rank`? If we only want only one rank to print, then we need to use `deviceId` instead